hw3/hw3-react/src/views/Search.js

```diff
@@ -0,0 +1,41 @@
+import { useState } from "react";
+
+export default function Search(prop) {
+  const [input, setInput] = useState("");
+  let fullName = [];
+  let title = [];
+  let imageUrl = [];
+  for (let i = 0; i < prop.character.family.length; i++) {
+    fullName[i] = prop.character.family[i].fullName;
+    title[i] = prop.character.family[i].title;
+    imageUrl[i] = prop.character.family[i].imageUrl;
+  }
+  //console.log(fullName, title, imageUrl);
+  function searchFunction() {
+    let { name } = "";
+    for (let i = 0; i < fullName.length; i++) {
+      if (input === "") return "Please Enter Name";
+      else if (fullName[i].toLowerCase() === input.toLowerCase()) {
+        name = fullName[i];
+      }
+      if (name !== "") return name;
+      else return "Not found";
+    }
```

Comment:
I think it might be helpful to replace the `for` loop with an array `filter()`, since you're trying to find an `input` match in `fullName`.  

Then, you could work with the resulting filtered array to either display any matches or return "not found".

Reply:
Thank you will try that out.